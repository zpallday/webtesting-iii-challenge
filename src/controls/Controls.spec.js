  
import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Controls from './Controls'


describe('<Control />', () => {

    it('matches snapshot', () => {
        const tree = renderer.create(<Controls />);
    
 
        expect(tree.toJSON()).toMatchSnapshot();
      })

it ('should render the toggle and btn', () => {
    const { getByText } = render (<Controls/>)
    getByText(/close gate/i)
    getByText(/lock gate/i)
})

it( 'should disable lock/unlock btn', () => {
    const { getByText } = render(<Controls closed={false} />)
    const btn = getByText(/lock gate/i)
    expect(btn).toBeDisabled(); 
})

it('should disable open/ closed btn locked', () => {
    const {getByText} = render(<Controls locked={true} />)
        const btn = getByText(/close gate/i)
        expect(btn).toBeDisabled()
    })
 it(' should toggleCloseed on btn clicked', () => {
     const toggleClosed = jest.fn();
     const { getByText } = render(<Controls toggleClosed={toggleClosed} />)
     const btn = getByText(/close gate/i)
     fireEvent.click(btn)
     expect(toggleClosed).toBeCalledTimes(1)
 })
})




    

