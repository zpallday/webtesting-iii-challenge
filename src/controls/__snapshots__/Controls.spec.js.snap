// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Control /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;
